tests/blackbox_test.go

```diff
@@ -0,0 +1,1215 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package blackbox
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"testing"
+)
+
+type File struct {
+	Name     string
+	Path     string
+	Contents []string
+	Mode     string
+}
+
+type Partition struct {
+	Number         int
+	Label          string
+	TypeCode       string
+	TypeGUID       string
+	GUID           string
+	Device         string
+	Offset         int
+	Length         int
+	FilesystemType string
+	MountPath      string
+	Hybrid         bool
+	Files          []File
+}
+
+type MntDevice struct {
+	label string
+	code  string
+}
+
+type Test struct {
+	name       string
+	in         []*Partition
+	out        []*Partition
+	mntDevices []MntDevice
+	config     string
+}
+
+func getBaseDisk() []*Partition {
+	return []*Partition{
+		{
+			Number:         1,
+			Label:          "EFI-SYSTEM",
+			TypeCode:       "efi",
+			Length:         262144,
+			FilesystemType: "ext4",
+			Hybrid:         true,
+			Files: []File{
+				{
+					Name:     "multiLine",
+					Path:     "path/example",
+					Contents: []string{"line 1", "line 2"},
+				}, {
+					Name:     "singleLine",
+					Path:     "another/path/example",
+					Contents: []string{"single line"},
+				}, {
+					Name: "emptyFile",
+					Path: "empty",
+				}, {
+					Name: "noPath",
+					Path: "",
+				},
+			},
+		}, {
+			Number:   2,
+			Label:    "BIOS-BOOT",
+			TypeCode: "bios",
+			Length:   4096,
+		}, {
+			Number:         3,
+			Label:          "USR-A",
+			GUID:           "7130c94a-213a-4e5a-8e26-6cce9662f132",
+			TypeCode:       "coreos-rootfs",
```

Comment:
I followed what was defined in https://github.com/coreos/scripts/blob/master/build_library/disk_layout.json#L29